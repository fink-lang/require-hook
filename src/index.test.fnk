{describe, it, mock, mock_fn, snapshot, eq, match_any} = import `./testing`

mock_hooks = []
mock_maps = []


mock:: `pirates`, fn: {
  addHook: mock_fn:: fn ...args: mock_hooks.push(args)
}

mock:: `source-map-support`, fn: {
  install: mock_fn:: fn ...args: mock_maps.push(args)
}

mock:: `@fink/loxia`, fn: {
  generate: mock_fn()
}

{generate} = import `@fink/loxia`


describe:: `hook`, fn:
  require(`.`)

  it:: `registers hook`, fn:
    mock_hooks eq [
      [match_any(Function), {exts: [`.fnk`], ignoreNodeModules: false}]
    ]


  it:: `transforms fnk`, fn:
    generate.mockReturnValue({code: `test-code`, map: `test-map`})

    [[hook]] = mock_hooks

    hook(`test = 1234`, `test.fnk`) eq `test-code`


  it:: `registers source map support`, fn:
    mock_maps eq [[{
      handleUncaughtExceptions: false,
      environment: `node`,
      retrieveSourceMap: match_any(Function)
    }]]


  it:: `gets source-map`, fn:
    [[{retrieveSourceMap}]] = mock_maps

    retrieveSourceMap(`test.fnk`) eq {map: `test-map`, url: null}


  it:: `returns null for missing source-map`, fn:
    [[{retrieveSourceMap}]] = mock_maps

    retrieveSourceMap(`test-foo.fnk`) eq null

