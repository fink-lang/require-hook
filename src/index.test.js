import {addHook} from 'pirates';


jest.mock('pirates', ()=> ({
  addHook: jest.fn()
}));


describe('hook', ()=> {
  it('registers upon import', ()=> {
    const {transform} = require('.');

    expect(addHook).toHaveBeenCalledWith(
      transform, {exts: ['.fnk'], ignoreNodeModules: false}
    );
  });


  it('transforms fnk', ()=> {
    const {transform} = require('.');

    expect(
      transform('foobar = 3', 'foobar.fnk')
    ).toBe(
      'const foobar = 3;'
    );
  });
});


