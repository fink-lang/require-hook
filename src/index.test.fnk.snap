// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hook gets source-map 1`] = `
Object {
  "map": Object {
    "mappings": "MAAA,I,GAAO,I;QAAP,I,GAAA,I",
    "names": Array [],
    "sources": Array [
      "test.fnk",
    ],
    "sourcesContent": Array [
      "test = 1234",
    ],
    "version": 3,
  },
  "url": null,
}
`;

exports[`hook transforms fnk 1`] = `
"const test = 1234;
exports.test = test;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInRlc3QuZm5rIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJNQUFBLEksR0FBTyxJO1FBQVAsSSxHQUFBLEkiLCJzb3VyY2VzQ29udGVudCI6WyJ0ZXN0ID0gMTIzNCJdfQ=="
`;
